service/storage_proxy.cc

```diff
@@ -330,6 +331,43 @@ class cas_mutation : public mutation_holder {
     }
 };
 
+// Tracks the success of a set of CDC mutations generated from the same mutation
+// Used to updated CDC-related metrics
+struct cdc_augmentation_tracker {
+private:
+    ::cdc::stats& _stats;
+    bool _had_error;
+    bool _was_split;
+    ::cdc::stats::part_type_set _touches;
+    utils::latency_counter _lc;
+
+public:
+    cdc_augmentation_tracker(::cdc::stats& s, bool was_split, ::cdc::stats::part_type_set touches, utils::latency_counter lc)
+        : _stats(s)
+        , _had_error(false)
+        , _was_split(was_split)
+        , _touches(touches)
+        , _lc(lc)
+    {}
+
+    ~cdc_augmentation_tracker() {
+        auto& histograms = _had_error ? _stats.histograms_failed : _stats.histograms_succeeded;
+        histograms.write_latency.add(_lc.stop().latency(), histograms.write_latency._count + 1);
+        if (_had_error) {
+            if (_was_split) {
+                _stats.counters_failed.split_count++;
```

Comment:
Here is an example of what I asked above... Something sets _was_split, and now we do split_count++. Why couldn't the split function just do the split_count++ itself?

Reply:
Because the place that splits does not know whether the operation will succeed or fail.